next.config.js

```diff
@@ -16,6 +20,44 @@ const temporary_hideDocsPaths = {
   has: [{ type: 'host', value: '(^(?!.*(waypointproject)).*$)' }],
 }
 
+/**
+ * Reads in config files from config/[env].json and replaces references in the code
+ * with the literal values using webpack.DefinePlugin
+ */
+function HashiConfigPlugin() {
+  const env = process.env.HASHI_ENV || 'development'
+  const envConfigPath = path.join(process.cwd(), 'config', `${env}.json`)
+
+  function getHashiConfig(path) {
+    try {
+      const envConfig = JSON.parse(fs.readFileSync(path))
+      const ret = flat(envConfig, {
+        safe: true,
+      })
+      return ret
+    } catch (err) {
+      console.log('Error loading environment config:', err)
+      return {}
+    }
+  }
+
+  return new webpack.DefinePlugin({
+    ...Object.fromEntries(
+      Object.entries(getHashiConfig(envConfigPath)).map(([key]) => {
+        return [
+          `config.${key}`,
+          webpack.DefinePlugin.runtimeValue(
+            () => {
+              return JSON.stringify(getHashiConfig(envConfigPath)[key])
+            },
+            { fileDependencies: [envConfigPath] }
+          ),
```

Comment:
This allows the values to update during development. It's a little thing, but a nice QOL improvement for iterating locally.

(note: new values won't get added, but existing values will update when the config file changes)